packages/stark-core/src/common/bootstrap/abstract-stark-main.ts

```diff
@@ -0,0 +1,163 @@
+import { ApplicationRef, ComponentRef, enableProdMode, NgModuleRef } from "@angular/core";
+import { disableDebugTools, enableDebugTools } from "@angular/platform-browser";
+import { createNewHosts } from "@angularclass/hmr";
+
+import { StarkMain } from "./stark-main.intf";
+import {StarkEnvironment} from "../environment";
+
+/**
+ * Parent class for bootstrapping Stark applications.
+ * It handles the DOMContentLoaded event of the Web browser in order to bootstrap the application.
+ * This class also supports HMR in development mode.
+ */
+export abstract class AbstractStarkMain implements StarkMain {
+	protected constructor(protected environment: StarkEnvironment) {
+		// no-op
+	}
+
+	/**
+	 * Must be implemented by children classes. Called within bootstrapDomReady (i.e., when the DOM has been loaded)
+	 */
+	public abstract main: () => Promise<any>;
+
+	public mainWrapper = (): void => {
+		// written like this because otherwise "this" will not be captured :)
+		// no need for the event listener after this
+		document.removeEventListener("DOMContentLoaded", this.mainWrapper, false);
+
+		this.initialize();
+	};
+
+	/**
+	 * Call the main() method and log an error in case of failure
+	 */
+	private invokeMain(): void {
+		// FIXME handle errors nicely (UX!)
+		this.main().catch((err: any) => console.error(err));
+
+		// TODO reintroduce later
+		// bootstrap the app as long as the app root element is there (it won't be there in case the browser is outdated)
+		// if (document.getElementsByClassName("stark-app")[0]) {
+		// 	// add anything relevant here
+		// } else {
+		// 	console.error("Could not bootstrap the App: unsupported browser.");
+		// }
+	}
+
+	/**
+	 * Initialize by configuring HMR if it is enabled or normally instead.
+	 */
+	protected initialize = (): void => {
```

Comment:
same here about the definition of the method

Reply:
Same reply